### PR TITLE
Send `400 Bad Request` when receiving an invalid SAML request

### LINKED ISSUE #379: 'Returning an internal server error: failed to get the value of a parameter: SAMLResponse'
We sometimes get the following warning log:

```
[id: 0x1172fb1d, L:... - R:...][h2://.../saml/acs/post#GET] Returning an 
internal server error: failed to get the value of a parameter: SAMLResponse
om.linecorp.armeria.server.saml.SamlException: failed to get the value of a 
parameter: SAMLResponse
```

Should we respond with other status code, such as `400 Bad Request`?

### PR DESCRIPTION
**Pull request by hyangtack**
Modifications:
- Update Armeria: 0.86.0 -> 0.87.0
- Send `400 Bad Request` for `InvalidSamlRequestException`.

Result:
- Closes #379

### CODE REVIEW

**Comment by hyangtack** [comment]
Fixed [here](https://github.com/line/centraldogma/pull/406/files#diff-f03bc11d746f55feb6abaf793ff90d5aR177) for #379. The other changes are for updating Armeria. :-)

File: server/src/main/java/com/linecorp/centraldogma/server/auth/AuthProvider.java [hunk 1]
```diff
@@ -66,20 +66,20 @@
...
 
     /**
-     * A set of {@link PathMapping}s which handles a login request. It is necessary only if
+     * A set of {@link Route}s which handles a login request. It is necessary only if
      * an authentication protocol requires a login feature provided by the server.
      */
-    Set<PathMapping> LOGIN_API_PATH_MAPPINGS =
-            ImmutableSet.of(PathMapping.ofExact(API_V0_PATH_PREFIX + "authenticate"),
-                            PathMapping.ofExact(API_V1_PATH_PREFIX + "login"));
+    Set<Route> LOGIN_API_PATH_MAPPINGS =
```
Comment (minwoox) [inline server/src/main/java/com/linecorp/centraldogma/server/auth/AuthProvider.java]: nit: `LOGIN_API_ROUTES`?

File: server/src/main/java/com/linecorp/centraldogma/server/CentralDogma.java [hunk 2]
```diff
@@ -704,7 +704,7 @@ protected HttpResponse doGet(ServiceRequestContext ctx, HttpRequest req) {
             final Service<HttpRequest, HttpResponse> logout =
                     Optional.ofNullable(authProvider.logoutApiService())
                             .orElseGet(() -> new DefaultLogoutService(executor));
-            for (PathMapping mapping : LOGOUT_API_PATH_MAPPINGS) {
+            for (Route mapping : LOGOUT_API_PATH_MAPPINGS) {
```
Comment (minwoox) [inline server/src/main/java/com/linecorp/centraldogma/server/CentralDogma.java]: nit: mapping -> route?

**Comment by trustin** [comment]
Thanks, @hyangtack!

**Pull request merged**